src/cmds/scripts/pbs_habitat.in

```diff
@@ -221,8 +221,9 @@ upgrade_pbs_database() {
 		return 1
 	fi
 	
-	sys_pgsql_ver=$(echo `${PGSQL_DIR}/bin/psql -V` | awk 'NR==1 {print $NF}' | cut -d '.' -f 1,2)
+	sys_pgsql_ver=$(echo `${PGSQL_BIN}/postgres -V` | awk 'NR==1 {print $NF}' | cut -d '.' -f 1,2)
 	old_pgsql_ver=`cat ${data_dir}/PG_VERSION`
+	[[ $old_pgsql_ver != *"."* ]] && sys_pgsql_ver=$(echo $sys_pgsql_ver | cut -d '.' -f 1)
```

Comment:
Just a suggestion. Instead of using the regular expression which looks like greedy can we use something like [[ ! "$old_pgsql_ver" =~ "." ]]